grpc-proto/src/main/java/io/stargate/grpc/Values.java

```diff
@@ -163,4 +165,141 @@ public static Value of(Map<Value, Value> fields) {
+
+  public static short smallint(Value value) {
+    checkInnerCase(value, InnerCase.INT);
+
+    short shortValue = (short) value.getInt();
+    if (shortValue != value.getInt()) {
+      throw new IllegalArgumentException(
+          String.format("Valid range for smallint is %d to %d", Short.MIN_VALUE, Short.MAX_VALUE));
+    }
+    return shortValue;
+  }
+
+  public static byte tinyint(Value value) {
+    checkInnerCase(value, InnerCase.INT);
+
+    byte byteValue = (byte) value.getInt();
+    if (byteValue != value.getInt()) {
+      throw new IllegalArgumentException(
+          String.format("Valid range for tinyint is %d to %d", Byte.MIN_VALUE, Byte.MAX_VALUE));
+    }
+    return byteValue;
+  }
+
+  public static float float_(Value value) {
+    checkInnerCase(value, InnerCase.FLOAT);
+
+    return value.getFloat();
+  }
+
+  public static double double_(Value value) {
+    checkInnerCase(value, InnerCase.DOUBLE);
+
+    return value.getDouble();
+  }
+
+  public static ByteBuffer byteBuffer(Value value) {
+    checkInnerCase(value, InnerCase.BYTES);
+
+    ByteBuffer bytes = ByteBuffer.allocate(value.getBytes().size());
+    value.getBytes().copyTo(bytes);
+    return bytes;
+  }
+
+  public static byte[] bytes(Value value) {
+    checkInnerCase(value, InnerCase.BYTES);
+
+    return value.getBytes().toByteArray();
+  }
+
+  public static String string(Value value) {
+    checkInnerCase(value, InnerCase.STRING);
+
+    return value.getString();
+  }
+
+  public static UUID uuid(Value value) {
+    checkInnerCase(value, InnerCase.UUID);
+
+    if (value.getUuid().getValue().size() != 16) {
+      throw new IllegalArgumentException("Expected 16 bytes for a uuid values");
+    }
+    ByteBuffer bytes = ByteBuffer.allocate(16);
+    value.getUuid().getValue().copyTo(bytes);
+    return new UUID(bytes.getLong(0), bytes.getLong(8));
+  }
+
+  public static InetAddress inet(Value value) {
+    checkInnerCase(value, InnerCase.INET);
+
+    int size = value.getInet().getValue().size();
+    if (size != 4 && size != 16) {
+      throw new IllegalArgumentException(
+          "Expected 4 bytes (IPv4) or 16 (IPv6) bytes for a inet values");
+    }
+    try {
+      return InetAddress.getByAddress(value.getInet().getValue().toByteArray());
+    } catch (UnknownHostException e) {
+      throw new IllegalArgumentException("Invalid bytes in inet value");
+    }
+  }
+
+  public static BigInteger varint(Value value) {
+    checkInnerCase(value, InnerCase.VARINT);
+
+    return new BigInteger(value.getVarint().getValue().toByteArray());
+  }
+
+  public static BigDecimal decimal(Value value) {
+    checkInnerCase(value, InnerCase.DECIMAL);
+
+    return new BigDecimal(
+        new BigInteger(value.getDecimal().getValue().toByteArray()), value.getDecimal().getScale());
+  }
+
+  public static LocalDate date(Value value) {
+    checkInnerCase(value, InnerCase.DATE);
+
+    int unsigned = value.getDate();
+    int signed = unsigned + Integer.MIN_VALUE;
```

Comment:
just for curiosity, why do we need `+ Integer.MIN_VALUE` and `- Integer.MIN_VALUE;` for another way around? Can't we drop this addition/substraction?

Reply:
This is the representation of date:

```
    // CQL types: date
    // An unsigned integer representing days with Unix epoch (January, 1 1970) at 2^31.
    // Examples:
    // 0:    -5877641-06-23
    // 2^31: 1970-1-1
    // 2^32: 5881580-07-11
    uint32 date = 11;
```